src/matrix/kernelmatrix.jl

```diff
@@ -8,74 +8,82 @@ kernelmatrix!
 
 function kernelmatrix!(
     K::AbstractMatrix,
-    kernel::Kernel,
-    X::AbstractVector;
+    κ::SimpleKernel,
+    X::AbstractMatrix;
     obsdim::Int = defaultobs,
 )
-    return kernelmatrix!(K, kernel, reshape(X, 1, :); obsdim = 2)
+    @assert obsdim ∈ [1, 2] "obsdim should be 1 or 2 (see docs of `kernelmatrix`)"
+    if !check_dims(K, X, X, feature_dim(obsdim), obsdim)
+        throw(DimensionMismatch("Dimensions of the target array K $(size(K)) are not consistent with X $(size(X))"))
+    end
+    map!(x -> kappa(κ, x), K, pairwise(metric(κ), X, dims = obsdim))
 end
 
 function kernelmatrix!(
-        K::AbstractMatrix,
-        κ::Kernel,
-        X::AbstractMatrix;
-        obsdim::Int = defaultobs
-        )
-        @assert obsdim ∈ [1,2] "obsdim should be 1 or 2 (see docs of kernelmatrix))"
-        if !check_dims(K,X,X,feature_dim(obsdim),obsdim)
-            throw(DimensionMismatch("Dimensions of the target array K $(size(K)) are not consistent with X $(size(X))"))
-        end
-        map!(x->kappa(κ,x),K,pairwise(metric(κ),X,dims=obsdim))
+    K::AbstractMatrix,
+    κ::Kernel,
+    X::AbstractMatrix;
+    obsdim::Int = defaultobs
+)
+    @assert obsdim ∈ [1, 2] "obsdim should be 1 or 2 (see docs of `kernelmatrix`))"
+    if obsdim == 1
+        kernelmatrix!(K, κ, ColVecs(X'))
+    else
+        kernelmatrix!(K, κ, ColVecs(X))
+    end
```

Comment:
Now that we have `vec_of_vecs` and RowVecs this could be changed to
```suggestion
    kernelmatrix!(K, κ, vec_of_vecs(X; obsdim = obsdim))
```